bin/comment_with_dependency_diff

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -euo pipefail
+
+MODULE="${1:-}"
+CONFIGURATION="${2:-}"
+
+if [ -z "$MODULE" ] || [ -z "$CONFIGURATION" ]; then
+  echo "Not enough arguments provided. Usage: ./comment_with_dependency_diff <MODULE> <CONFIGURATION>"
+  exit 1
+fi
+
+OWNER=$(basename "$(dirname "${BUILDKITE_PULL_REQUEST_REPO}")")
+REPO=$(basename "${BUILDKITE_PULL_REQUEST_REPO%.git}")
+
+DIFF_DEPENDENCIES_FOLDER="./build/reports/diff"
+BASE_BRANCH_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/base_branch_dependencies.txt"
+HEAD_BRANCH_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/head_branch_dependencies.txt"
+BASE_BRANCH_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/base_branch_build_env.txt"
+HEAD_BRANCH_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/head_branch_build_env.txt"
+DIFF_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/diff_dependencies.txt"
+DIFF_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/diff_build_env.txt"
+DEPENDENCY_TREE_VERSION="1.2.1"
+
+PR_NUMBER=$BUILDKITE_PULL_REQUEST
+PR_URL="https://api.github.com/repos/$OWNER/$REPO/pulls/$PR_NUMBER"
+
+git config --global user.email "mobile@automattic.com"
+git config --global user.name "Dependency Tree Diff Tool"
+
+echo "--> Starting the check"
+echo "--> Fetching the base branch from $PR_URL"
+githubResponse="$(curl "$PR_URL" -H "Authorization: token $GITHUB_TOKEN")"
+baseBranch=$(echo "$githubResponse" | tr '\r\n' ' ' | jq '.base.ref' | tr -d '"')
+echo "--> Base branch is $baseBranch"
+
+git merge "origin/$baseBranch" --no-edit
+
+if (git diff --name-status "origin/$baseBranch" | grep -E "\.gradle(\.kts)?|\.toml" --quiet); then
+    echo ".gradle files have been changed. Looking for caused dependency changes"
+else
+    echo ".gradle files haven't been changed. There is no need to run the diff"
+    exit 0
+fi
+
+mkdir -p "$DIFF_DEPENDENCIES_FOLDER"
+
+echo "--> Generating dependencies to the file $HEAD_BRANCH_DEPENDENCIES_FILE"
+./gradlew :"$MODULE":dependencies --configuration "$CONFIGURATION" >$HEAD_BRANCH_DEPENDENCIES_FILE
+
+echo "--> Generating build environment dependencies for the head branch"
+./gradlew buildEnvironment >$HEAD_BRANCH_BUILD_ENV_FILE
+
+echo "--> Generating dependencies to the file $BASE_BRANCH_DEPENDENCIES_FILE"
+
+if ! git diff --quiet; then
+  echo "Changes detected. Stashing and continuing with the script"
+  git diff
+  git stash
+fi
+
+git checkout "$baseBranch"
+./gradlew :"$MODULE":dependencies --configuration "$CONFIGURATION" >$BASE_BRANCH_DEPENDENCIES_FILE
+
+echo "--> Generating build environment dependencies for the base branch"
+./gradlew buildEnvironment >$BASE_BRANCH_BUILD_ENV_FILE
+
+git checkout "-"
+
+echo "--> Downloading dependency-tree-diff.jar"
+curl -v -L "https://github.com/JakeWharton/dependency-tree-diff/releases/download/$DEPENDENCY_TREE_VERSION/dependency-tree-diff.jar" -o dependency-tree-diff.jar
+sha=$(sha1sum dependency-tree-diff.jar | cut -d' ' -f1)
+if [[ $sha != "39c50f22cd4211a3e52afec43120b7fdf90f9890" ]]; then
```

Comment:
💡  Nitpick: could be worth extracting this SHA in a `DEPENDENCY_TREE_CHECKSUM` constant to put next to line 23, so that it'd be easier to update to a new version in the future by changing the version and checksum together?